tests/test_allow_none_vs_default.py

```diff
@@ -0,0 +1,191 @@
+"""Copyright 2019 Equinor ASA and The Netherlands Organisation for
+Applied Scientific Research TNO.
+
+Licensed under the MIT license.
+
+Permission is hereby granted, free of charge, to any person obtaining a
+copy of this software and associated documentation files (the
+"Software"), to deal in the Software without restriction, including
+without limitation the rights to use, copy, modify, merge, publish,
+distribute, sublicense, and/or sell copies of the Software, and to
+permit persons to whom the Software is furnished to do so, subject to
+the conditions stated in the LICENSE file in the project root for
+details.
+
+The above copyright notice and this permission notice shall be included
+in all copies or substantial portions of the Software.
+"""
+
+
+import unittest
+
+
+import configsuite
+from configsuite import MetaKeys as MK
+from configsuite import types
+
+
+class TestNotAllowNoneVsDefault(unittest.TestCase):
```

Comment:
I wonder if it would be good to separate this a bit. How about having two classes;

`class TestAllowNoneCombineDefaultRequired`, `class TestDisallowNoneCombineDefaultRequired`

Reply:
I don't mind ;) With https://github.com/equinor/configsuite/issues/95 the `required` option might be deprecated and removed anyway... Removing the right tests based on this will be quicker if split though.

Reply:
I can do that, similarly I can split it in two based on `AllowNone` or not `AllowNone` or on having a default value. The point of this PR is to ensure testing of all 8 combinations and hence, after pondering a bit I see it as most natural to keep them in the same class. Let me know if you disagree :)